api/app/src/providers/withings.ts

```diff
@@ -79,6 +76,60 @@ export class Withings extends Provider implements OAuth2 {
     return JSON.stringify(response.data.body);
   }
 
+  async getAccessToken(connectedUser: ConnectedUser): Promise<string> {
+    const providerData = getProviderDataFromConnectUserOrFail(connectedUser, PROVIDER_WITHINGS);
+
+    const token = providerData.token;
+
+    const refreshedToken = await this.checkRefreshToken(token, connectedUser);
+
+    return refreshedToken.access_token;
+  }
+
+  async checkRefreshToken(token: string, connectedUser: ConnectedUser): Promise<Token> {
```

Comment:
The getAccessToken specifically for withings could be cleaner and refactored with whats in oauth2 just trying to push it out quickly but let me know and i can clean it up

Reply:
All good for now